### PR TITLE
Fix: Handle all modifiers in lexical preserving printer #2340

### LINKED ISSUE #2340: Handle all modifiers in lexical preserving printer
#2339 - check that all modifiers are handled.

### PR DESCRIPTION
**Pull request by jlerbsc**
Fixes #2340 Handle all modifiers in lexical preserving printer

### CODE REVIEW

File: javaparser-core/src/main/java/com/github/javaparser/printer/lexicalpreservation/Difference.java [hunk 1]
```diff
@@ -387,7 +392,7 @@ private void applyRemovedDiffElement(RemovedGroup removedGroup, Removed removed,
         } else if (originalElementIsToken && originalElement.isWhiteSpaceOrComment()) {
             originalIndex++;
         } else if (removed.isPrimitiveType()) {
-            if (isPrimitiveType(originalElement)) {
+            if (originalElement.isPrimitive()) {
```
Comment (matozoid) [inline javaparser-core/src/main/java/com/github/javaparser/printer/lexicalpreservation/Difference.java:395]: Nice!

File: javaparser-core-testing/src/test/java/com/github/javaparser/printer/lexicalpreservation/PrettyPrinterIssue2340Test.java [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+import com.github.javaparser.ast.body.EnumDeclaration;
+
+class PrettyPrinterIssue2340Test extends AbstractLexicalPreservingTest {
+
+    @Test
+    void printingVariableDeclarationWithAddedModifier() {
+        String def2 = "List i";
+        considerVariableDeclaration(def2);
+        expression.asVariableDeclarationExpr().getModifiers().addFirst(Modifier.privateModifier());
+//        System.out.println(LexicalPreservingPrinter.print(expression));
```
Comment (matozoid) [inline javaparser-core-testing/src/test/java/com/github/javaparser/printer/lexicalpreservation/PrettyPrinterIssue2340Test.java]: Do you want to keep the println's in here?

File: javaparser-core/src/main/java/com/github/javaparser/printer/lexicalpreservation/Difference.java [hunk 3]
```diff
@@ -495,6 +515,39 @@ private void applyKeptDiffElement(Kept kept, TextElement originalElement, boolea
...
+
+    /*
+     * Returns the number of tokens to skip in originalElements list to synchronize with the DiffElements list
+     * This is due to the fact that types are considered as token in the originalElements list.
+     * For example,
+     * List<String> is represented by 4 tokens ([List][<][String][>]) while it's a CsmChild element in the DiffElements list
+     * So in this case, getIndexToNextTokenElement(..) returns 3 because we have to skip 3 tokens ([<][String][>]) to synchronize
+     * DiffElements list and originalElements list 
+     * The end of recursivity is reached when there is no next token or if the next token is a whitespace,EOL or tab, to take into account this type of declaration
+     * List<List<String>> l
```
Comment (matozoid) [inline javaparser-core/src/main/java/com/github/javaparser/printer/lexicalpreservation/Difference.java]: This doesn't take this one into account, I think: `List      <String> a;`. It is an overall improvement though (although a solid algorithm is preferred)

Comment (jlerbsc) [reply javaparser-core/src/main/java/com/github/javaparser/printer/lexicalpreservation/Difference.java]: You are right!

File: javaparser-core-testing/src/test/java/com/github/javaparser/printer/lexicalpreservation/PrettyPrinterIssue2340Test.java (hunk 2 above)
Comment (jlerbsc) [reply javaparser-core-testing/src/test/java/com/github/javaparser/printer/lexicalpreservation/PrettyPrinterIssue2340Test.java]: No it's useless.

File: javaparser-core/src/main/java/com/github/javaparser/printer/lexicalpreservation/Difference.java (hunk 3 above)
Comment (matozoid) [reply javaparser-core/src/main/java/com/github/javaparser/printer/lexicalpreservation/Difference.java]: Does that mean you're going to improve it, or that you consider this a good enough improvement? ;-)

Comment (jlerbsc) [reply javaparser-core/src/main/java/com/github/javaparser/printer/lexicalpreservation/Difference.java]: I have pushed an improvment which take into account your remark.

**Comment by matozoid** [comment]
Thanks!

**Pull request merged**